From 49eeb0b61eb2e817d90a2b8ef98ba9236350deaa Mon Sep 17 00:00:00 2001
From: Pavel Grunt <pgrunt@redhat.com>
Date: Tue, 14 Apr 2015 17:06:35 +0200
Subject: [PATCH] virt-viewer-main: '--wait' should not be used without domain
 name

Commit a830275344c88aef12166661b68ea2b4429c7212 required the domain
name to be placed just after the '--wait' option. It broke the
command line api, because running 'virt-viewer $vm --wait' was considered
as the error.

This patch rather checks whether the domain name was specified.

Related: rhbz#1209398, rhbz#1211573
(cherry picked from commit 69a5077a8ab99c3c83d73d81fbe1f29a46d4fdb8)
---
 src/virt-viewer-main.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/virt-viewer-main.c b/src/virt-viewer-main.c
index 3fae955..81d7c64 100644
--- a/src/virt-viewer-main.c
+++ b/src/virt-viewer-main.c
@@ -109,6 +109,11 @@ int main(int argc, char **argv)
         goto cleanup;
     }
 
+    if (args == NULL && waitvm) {
+        g_printerr(_("\nNo DOMAIN-NAME|ID|UUID was specified for '--wait'\n\n"));
+        goto cleanup;
+    }
+
     viewer = virt_viewer_new(uri, (args) ? args[0] : NULL, direct, attach, waitvm, reconnect);
     if (viewer == NULL)
         goto cleanup;
